pandas/io/json/_normalize.py

```diff
@@ -305,8 +306,24 @@ def _recursive_extract(data, path, seen_meta, level=0):
                 # For repeating the metadata later
                 lengths.append(len(recs))
                 for val, key in zip(meta, meta_keys):
+                    # Extract the value of the key when the level
+                    # is at the meta path end
                     if level + 1 > len(val):
                         meta_val = seen_meta[key]
+                        meta_vals[key].append(meta_val)
+                    # Extract the value of the key from seen_meta when
+                    # meta path and record path are on two branches
+                    elif seen_meta:
+                        meta_val = seen_meta[key]
+                        meta_vals[key] += [
+                            # The list case
```

Comment:
I think this entire routine would be much more clear if you pull out the ```meta_vals[key].append(meta_val) out a level, and just have each if/else set meta_val

Reply:
Thanks for the suggestion. Definitely!